### ISSUE TITLE
Application menu, pref property boolean `show_in_menu`

### ISSUE THREAD

**Issue by RobertKosten** [QubesOS/qubes-issues#6693]
**The problem you're addressing (if any)**
Similar to #4005 I find the dom0 Application menu very cluttered, since I prefer to create AppVMs for individual Apps where possible and, of course, base those off of individual templates (So, one `email-template`, with a couple `email-address-1`, etc. AppVMs).

**Describe the solution you'd like**
I don't think it needs to be as extensive as desired in other requests, a simple pref property `show_in_menu`, possibly even without GUI, would suffice.

**Where is the value to a user, and who might that user be?**
It would allow hiding all those `Template` and `Service` entries from the menu that rarely need accessing (and they still are accessible via the Qube manager, after all) and thus help declutter.

**Comment by DemiMarie** [comment]
The `internal` property is already for this, but it is not well documented if it is documented at all.

**Comment by andrewdavidwong** [comment]
> The `internal` property is already for this, but it is not well documented if it is documented at all.

There's already an issue for that: #6604.

It seems pretty clear to me that "internal" is _not_ intended merely for hiding a VM from the menu, but I guess we won't know for sure until #6604 is resolved.

**Comment by ninavizz** [comment]
@DemiMarie Does the `internal` property that already exists, allow the backend to control whether or not a VM shows-up in the AppMenu? I would LOVE to make that a GUI-available option for folks, in the new appmenu. Regardless, it needs to be better explained in our docs. I will head-scratch around how to best approach that, within the technical-documentation guidelines @andrewdavidwong has developed for curating today's qubes docs. :) 

What should Acceptance Criteria be to close this issue?
- Explain `internal` property in docs, perhaps in a "User Interface" section? There is already a basic mention of the app menu that I don't feel really provides much benefit beyond what most folks can figure out. It'd be nice to see documentation around how Qubes and XFCE "play together," such as this—and notation around how to create Favorites with the CLI
- Also: The broader concern will be resolved with #6665, which is the solution being developed from the design and user research work in #5677

**Comment by andrewdavidwong** [comment]
> What should Acceptance Criteria be to close this issue?
> 
> * Explain `internal` property in docs, perhaps in a "User Interface" section?

No. As I said, there's already an issue for that: https://github.com/QubesOS/qubes-issues/issues/6604

IMHO, the fate of this issue depends on how the developers decide to resolve #6604.

If the devs decide that `internal` means nothing over and above "not included in app menu," then this issue can be closed as a duplicate of #6604 without any further action, as the desired thing already exists. However, as I stated above, I highly doubt that this is what the devs really meant by `internal`, in which case defining it this way would be a mistake. It seems clear that the devs intend `internal` to mean something like "internal component of the system that we don't want users to mess with." Merely hiding something from the app menu is not sufficient to communicate to users that they shouldn't be messing with it, especially if it's easy to mess with in other ways. See https://github.com/QubesOS/qubes-issues/issues/6604#issuecomment-840761516.

If the devs decide that `internal` means something over and above "not included in app menu," then the thing this issue is asking for does not yet exist, and the devs can decide whether it should exist and whether they'll create it.

**Comment by deeplow** [comment]
> Does the `internal` property that already exists, allow the backend to control whether or not a VM shows-up in the AppMenu?

@ninavizz Yes. Look for `default-mgmt-dvm`. It shows in the Qube Manager but not in the meny. If in a dom0 terminal you type `qvm-features default-mgmt-dvm | grep internal` you'll see `internal` is enabled.

**Comment by ben-grande** [comment]
> **Where is the value to a user, and who might that user be?**
> It would allow hiding all those `Template` and `Service` entries from the menu that rarely need accessing (and they still are accessible via the Qube manager, after all) and thus help declutter.

That was done. Appmenu has `APPS, TEMPLATES, SERVICES`.

- For a VM to appear in `SERVICES`, it requires `servicevm` feature.
- For a VM to be hidden from most GUI tools, it requires `internal`.
- To customize the VM menu items, there is `default-menu-items`, `menu-items` and `menu-favorites`.
